src/db.rs

```diff
@@ -0,0 +1,407 @@
+use bb8_postgres::bb8::Pool;
+use bb8_postgres::bb8::RunError;
+use bb8_postgres::tokio_postgres::tls::NoTls;
+use bb8_postgres::tokio_postgres::{Row, Transaction};
+use bb8_postgres::PostgresConnectionManager;
+use openssl::encrypt::{Decrypter, Encrypter};
+use openssl::pkey::{PKey, Private};
+use openssl::rsa::Padding;
+use postgres_types::{FromSql, ToSql};
+use std::include_str;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum BotDbError {
+    #[error(transparent)]
+    PoolError(#[from] RunError<bb8_postgres::tokio_postgres::Error>),
+    #[error(transparent)]
+    PgError(#[from] bb8_postgres::tokio_postgres::Error),
+    #[error(transparent)]
+    EncryptError(#[from] openssl::error::ErrorStack),
+    #[error(transparent)]
+    StringError(#[from] std::string::FromUtf8Error),
+    #[error("City not found")]
+    CityNotFoundError,
+}
+
+#[derive(Debug, Clone)]
+pub struct DbController {
+    pool: Pool<PostgresConnectionManager<NoTls>>,
```

Comment:
just a thought related to fang. Since this struct uses bb8's pool. and fang's postgres implementation uses the same  struct in the queue, I think we can make it public in the queue to allow workers to re-use the same connection pool.

Reply:
Should be done in Fang module